enginepb: add OmitInRangefeeds field to MVCCValueHeader

RaduBerinde opened the pull request:
This commit is just the proto change that adds the field.

Informs: #113634
Epic: CRDB-13169
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115288)
<!-- Reviewable:end -->

RaduBerinde committed:
enginepb: add OmitInRangefeeds field to MVCCValueHeader

This commit is just the proto change that adds the field.

Informs: #113634
Epic: CRDB-13169
Release note: None

RaduBerinde commented:
I looked around the code and I can't figure out what else is needed beyond just setting this field in `mvccPutInternal`, like below (and of course some tests around that). The intent resolving code looks like will naturally copy the field as needed. Am I missing something major?

```diff
--- a/pkg/storage/mvcc.go
+++ b/pkg/storage/mvcc.go
@@ -2594,6 +2594,9 @@ func mvccPutInternal(
        versionValue := MVCCValue{}
        versionValue.Value = value
        versionValue.LocalTimestamp = opts.LocalTimestamp
+       if opts.Txn != nil {
+               versionValue.OmitInRangefeeds = opts.Txn.OmitInRangefeeds
+       }
 ```

miraradeva reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

For populating the field, I also think `mvccPutInternal` is the only place to change. That function does exit early for successful inline writes and replayed writes, but neither of them is relevant for this change I think; inline writes are not transactional, and replays don't result in a new write. 

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115288#-:-NkVcH7EDmxTuWDXlcBc:b-sax9jj)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andyyang890, @jbowens, and @nicktrav)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

And thanks for confirming, I will get that change out as soon as possible.

bors r+
